src/main/java/io/harness/cf/client/api/AuthService.java

```diff
@@ -33,10 +33,16 @@ protected void runOneIteration() {
       stopAsync();
       log.info("Stopping Auth service");
     } catch (ConnectorException e) {
-      log.error(
-          "Exception while authenticating, retry in {} seconds, error: {}",
-          pollIntervalInSec,
-          e.getMessage());
+      if (e.shouldRetry()) {
+        log.error(
+            "Exception while authenticating, retry in {} seconds, error: {}",
+            pollIntervalInSec,
+            e.getMessage());
+      } else {
+        stopAsync();
+        log.error("Exception while authenticating", e);
+        callback.onFailure(e.getMessage());
```

Comment:
When we aren't retrying, we shut down the async and call the `onFailure` that already exists. (I assume this was the original intent of that onFailure anyways and it just didn't get wired up)